File: articles/linear-algebra-for-systems-modeling/articles/pivot-structure-and-solvability/calculators/r/pivot_structure_calculator.R
result <- data.frame(
  calculator = "pivot_structure_solvability_calculator",
  equation_count = 3,
  unknown_count = 3,
  pivot_columns = "0,1,2",
  free_columns = "none",
  coefficient_rank = 3,
  augmented_rank = 3,
  consistent = TRUE,
  solution_behavior = "unique solution",
  tolerance = 1.0e-10,
  warning = "Solvability depends on rank comparison; feasibility depends on modeling assumptions."
)

dir.create("outputs", recursive = TRUE, showWarnings = FALSE)
write.csv(result, "outputs/r_pivot_structure_calculator.csv", row.names = FALSE)
print(result)

File: articles/linear-algebra-for-systems-modeling/articles/pivot-structure-and-solvability/calculators/python/pivot_structure_calculator.py
from __future__ import annotations

import csv
import json
from pathlib import Path


def main() -> None:
    output_dir = Path("outputs")
    output_dir.mkdir(parents=True, exist_ok=True)

    result = {
        "calculator": "pivot_structure_solvability_calculator",
        "equation_count": 3,
        "unknown_count": 3,
        "pivot_columns": "0,1,2",
        "free_columns": "none",
        "coefficient_rank": 3,
        "augmented_rank": 3,
        "consistent": True,
        "solution_behavior": "unique solution",
        "tolerance": 1.0e-10,
        "warning": "Solvability depends on rank comparison; feasibility depends on modeling assumptions."
    }

    with (output_dir / "pivot_structure_calculator.json").open("w", encoding="utf-8") as handle:
        json.dump(result, handle, indent=2, sort_keys=True)

    with (output_dir / "pivot_structure_calculator.csv").open("w", newline="", encoding="utf-8") as handle:
        writer = csv.DictWriter(handle, fieldnames=list(result.keys()))
        writer.writeheader()
        writer.writerow(result)

    print(json.dumps(result, indent=2, sort_keys=True))


if __name__ == "__main__":
    main()
